readme.md
#百度 UEditor 雪球 fork

百度竟然不提供 git repo（本来是有的，但是后来又不用了），真是弱爆了。

于是只好自建个 repo 放上雪球的 fork。百度下载的源码包包含太多臃肿内容，于是只添加 _src 下内容。


CHANGELOG.TXT
Ueditor Change List
Version 1.2.0
1.远程图片抓取
3.源码模式下css进行了简写
4.增加了baidu图片搜索功能，搜索图片然后直接插入到编辑器中
5.重写了浮动工具栏，支持混乱模式下的工具栏滚动
6.服务器图片在线管理
7.word的本地图片取得寛高
8.附件上传
9.自动排版
10.优化了状态反射的方式，改为编辑器获得焦点才会触发，失去焦点不在触发状态查询
11.添加了上来就可以全屏的配置项
12.修正了focus()方法，执行编辑器实例的focus方法可以选回失去焦点之前的选区
13.优化了查询状态反射的性能
14.添加了contentchagne事件
15.重写了autoheight插件，去掉setInterval的方式，并且长高时不在跳动
16.插入视频，可以预览，并且界面加入了视屏搜索功能，并且可以插入视屏预览图到编辑器中
17.单元格属性编辑
18.ie下的截屏功能
19.加强了table的dialog功能
20.改进了autolink的效果，例如: dddhttp://www.baidu.com 回车,http://www.baidu.com也可以被匹配到了
21.文件上传提供flash源码
22.修改了行间距的展示方式
23.段间距变为段前距和段后距
24.提供了.net的事例代码
25.首页提供了功能选择生成下载的新功能
26.首页文档进行了改进
27.分页符可以删除
28.增强了表格的编辑功能

Version 1.1.8
1.避免了重复加载源码高亮的核心代码
2.修复了word粘贴table过滤出错问题
3.修复插入地图会出现style="undefined"的问题
4.优化了list，多个相邻的属性一直的list会合并
5.可以在列表中的一行里产生多行的效果（通过回车再回退操作），类似office的效果
6.添加自定义样式功能
7.修了在chrome下右键删除td里的图片会把整个td删除的问题
8.改进了不同的页面调用一个editor，URL问题
9.增加了颜色选择器的颜色
10.改进了提供的后台程序的安全性
11.代码高亮支持折行
12.改进了源码编辑模式下的性能(ie下),并且支持自动换行
13.修改了在destroy之后会在ie下报错的问题
14.给初始化容器name值，那么在后台取值的键值就是name给定的值，方便多实例在一个form下提交
15.支持插入script/style这样的标签
16.修复了列表里插入浮动图片，图片不占位问题
17.源码模式下,去掉了pre中的&nbsp;
18.完善了_example下的demo例子
19.base64的图片被过滤掉了

Version 1.1.7.3
1.支持图片相对路径模式
2.word粘贴首行缩进问题
3.添加了图片边距
4.提供了图片等比压缩时基准边选择配置的功能
5.dialog在某些页面不显示问题
6.添加了行内间距的调整
7.在editor实例下添加了destroy方法
8.全屏按钮位置不对的问题
9.iframe.css支持相对和绝对路径
10.修正了focus方法在ff下失效的问题
11.提供了对FF3.6的支持
12.添加了Shift+Enter软回车功能
10.统一了颜色rgb转成#


version 1.1.7.2
1.去掉了iframe.css 改为在editor_config.js中配置，避免css文件找不到的问题
2.给下拉菜单添加了默认的文字说明
3.Ueditor.css去掉了对外部页面css的影响
4.修正了ie9下，编辑器的高度不随着内容缩短的问题
5.修正了粘贴有时会出现粘贴失败的情况
6.修正了在ie下点击图片会出现js错误的问题
7.修正了在ie下选全部替换，回退，再替换会出现替换失败的问题
8.增加表情本地化模式，可在config中配置是否开启本地化
9.flash的多图片上传
10.支持了源码模式的下的代码高亮
11.增加插入代码支持的语言，改进了插入代码的展示效果
12.增加了字数统计
13.增加了对图片的排版操作
14.解决ie6和ie7下工具栏浮动时cpu占用过高的bug
15.优化了文本模式粘贴的效果
16.优化了word粘贴的效果
17.在word粘贴本地图片时添加引导上传功能
18.更好的ie9支持
19.优化首行缩进效果
20.使用script标签代替textarea标签作为编辑器容器，简化前后端转码的配置。
21.优化了路径配置，修正了1.1.7.1中需要修改多处路径的问题
22.增加了图片操作浮层的开关配置
23.同时支持网络图片和本地图片的等比缩放
24.优化了源码模式下的代码格式

version 1.1.6.1
1.去掉了iframe.css 改为在editor_config.js中配置，避免css文件找不到的问题
2.给下拉菜单添加了默认的文字说明
3.Ueditor.css去掉了对外部页面css的影响
4.修正了ie9下，编辑器的高度不随着内容缩短的问题
5.修正了粘贴有时会出现粘贴失败的情况
6.修正了在ie下点击图片会出现js错误的问题
7.修正了在ie下选全部替换，回退，再替换会出现替换失败的问题





version 1.1.6
1. 插入日期按钮现在使用tangram日历控件
2. table可再编辑
3. 粘贴excel表格的问题
4. ff下最大化和切换源码出现光标不能跟着键盘改变和不能切出输入法的问题
5. tab按键功能
6. 支持多级列表
7. 超链接可以在非ie下去除下划线
8. 字体，字号，在editor-config.js中可配置



version 1.1.5
1.右键的策略，只显示选区内可操作的条目
2.禁止elementpath还会留下边框问题
3.字体改为了px
4.插入分页符
5.整合浮动toolbar为autofloat插件
6.初始化的值会在第一次操作前清除，而且不在有延迟感
7.配置项都放到了editor-config.js中
8.修正了多实例的问题
9.插入iframe功能
10.粘贴过滤掉内容会有提示，没过滤任何内容不会出现提示
11.修正代码高亮的显示效果
12.list放弃原生改为手动实现，修正一系列原生的bug
13.初始给个textarea会把内容取出作为初始值
14.去掉了源码状态下冗余的table/td/pre的style属性
15.fixed剪切出去会带start/end
16.fixex源码模式下getContent内容不是新的
17.table加入了设置背景颜色和边框颜色



Version 1.1.4
1.  锚点
2.  首行缩进
3.  行间距
4.  右键菜单
5.  插入代码
6.  文件上传(php版本）
7.  修复一些bug

Version 1.1.3
1.  修复chrome下粘贴的bug
2.  自动转换office粘入的有序列表和无序列表
3.  插入图片不再等比缩放，显示原始大小

Version 1.1.2
1.  修正IE9下autoHeight插件会一直长高的问题
2.  增加对IE6下大写style属性的转换处理（现统一转换成小写）
3.  格式刷
4.  上下标互斥
5.  form提交的支持
6.  增加了focus属性，可以初始化时，设置是否编辑器获得焦点
7.  增加了下滑线，删除线按钮，去掉了原来的下拉框
8.  autolink支持，使非ie在输入链接时能自动加上a标签
9.  google地图支持
10. 修正了一些bug


Version 1.1
1.	修改了删除链接的机制，允许一次性删除多个超链接
2.	改变了目录结构，方便部署（大大减少了开发代码过程中需要引入的js数量）
3.	修正部分bug

Version 1.0 (2011-7-8)
1.	完成功能的开发


editor_config.js
/**
 *  ueditor完整配置项
 *  可以在这里配置整个编辑器的特性
 */

(function () {
    //这里你可以配置成ueditor目录在您网站中与根目录之间的相对路径或者绝对路径（指以http开头的绝对路径）
    //window.UEDITOR_HOME_URL可以在外部配置，这里就不用配置了
    //场景：如果你有多个页面使用同一目录下的editor,因为路径不同，所以在使用editor的页面上配置这个路径写在这个js外边
    //var URL = window.UEDITOR_HOME_URL || '../';
    var tmp = window.location.pathname,
            URL = window.UEDITOR_HOME_URL||tmp.substr(0,tmp.lastIndexOf("\/")+1).replace("_examples/","").replace("website/","");//这里你可以配置成ueditor目录在您网站的相对路径或者绝对路径（指以http开头的绝对路径）
    UEDITOR_CONFIG = {
        imagePath:URL + "server/upload/", //图片文件夹所在的路径，用于显示时修正后台返回的图片url！具体图片保存路径需要在后台设置。！important
        compressSide:0,                   //等比压缩的基准，确定maxImageSideLength参数的参照对象。0为按照最长边，1为按照宽度，2为按照高度
        maxImageSideLength:900,          //上传图片最大允许的边长，超过会自动等比缩放,不缩放就设置一个比较大的值
        relativePath:true,                //是否开启相对路径。开启状态下所有本地图片的路径都将以相对路径形式进行保存.强烈建议开启！

        filePath:URL + "server/upload/",  //附件文件夹保存路径
	    catchRemoteImageEnable:true,                                   //是否开启远程图片抓取
        catcherUrl:URL +"server/submit/php/getRemoteImage.php",             //处理远程图片抓取的地址
        localDomain:"baidu.com",                                        //本地顶级域名，当开启远程图片抓取时，除此之外的所有其它域名下的图片都将被抓取到本地
	    imageManagerPath:URL + "server/submit/php/imageManager.php",       //图片在线浏览的处理地址
        UEDITOR_HOME_URL:URL,                                          //为editor添加一个全局路径
        //工具栏上的所有的功能按钮和下拉框，可以在new编辑器的实例时选择自己需要的从新定义
        toolbars:[
            ['FullScreen', 'Source', '|', 'Undo', 'Redo', '|',
                'Bold', 'Italic', 'Underline', 'StrikeThrough', 'Superscript', 'Subscript', 'RemoveFormat', 'FormatMatch','AutoTypeSet', '|',
                'BlockQuote', '|', 'PastePlain', '|', 'ForeColor', 'BackColor', 'InsertOrderedList', 'InsertUnorderedList','SelectAll', 'ClearDoc', '|', 'CustomStyle',
                'Paragraph', '|','RowSpacingTop', 'RowSpacingBottom','LineHeight', '|','FontFamily', 'FontSize', '|',
                'DirectionalityLtr', 'DirectionalityRtl', '|', '', 'Indent', '|',
                'JustifyLeft', 'JustifyCenter', 'JustifyRight', 'JustifyJustify', '|',
                'Link', 'Unlink', 'Anchor', '|', 'ImageNone', 'ImageLeft', 'ImageRight', 'ImageCenter', '|', 'InsertImage', 'Emotion', 'InsertVideo', 'Attachment', 'Map', 'GMap', 'InsertFrame', 'PageBreak', 'HighlightCode', '|',
                'Horizontal', 'Date', 'Time', 'Spechars','SnapScreen', 'WordImage', '|',
                'InsertTable', 'DeleteTable', 'InsertParagraphBeforeTable', 'InsertRow', 'DeleteRow', 'InsertCol', 'DeleteCol', 'MergeCells', 'MergeRight', 'MergeDown', 'SplittoCells', 'SplittoRows', 'SplittoCols', '|',
                 'Print', 'Preview', 'SearchReplace','Help']
        ],
        //当鼠标放在工具栏上时显示的tooltip提示
        labelMap:{
            'anchor':'锚点', 'undo':'撤销', 'redo':'重做', 'bold':'加粗', 'indent':'首行缩进','snapscreen': '截图',
            'italic':'斜体', 'underline':'下划线', 'strikethrough':'删除线', 'subscript':'下标',
            'superscript':'上标', 'formatmatch':'格式刷', 'source':'源代码', 'blockquote':'引用',
            'pasteplain':'纯文本粘贴模式', 'selectall':'全选', 'print':'打印', 'preview':'预览',
            'horizontal':'分隔线', 'removeformat':'清除格式', 'time':'时间', 'date':'日期',
            'unlink':'取消链接', 'insertrow':'前插入行', 'insertcol':'前插入列', 'mergeright':'右合并单元格', 'mergedown':'下合并单元格',
            'deleterow':'删除行', 'deletecol':'删除列', 'splittorows':'拆分成行', 'splittocols':'拆分成列', 'splittocells':'完全拆分单元格',
            'mergecells':'合并多个单元格', 'deletetable':'删除表格', 'insertparagraphbeforetable':'表格前插行', 'cleardoc':'清空文档',
            'fontfamily':'字体', 'fontsize':'字号', 'paragraph':'段落格式', 'insertimage':'图片', 'inserttable':'表格', 'link':'超链接',
            'emotion':'表情', 'spechars':'特殊字符', 'searchreplace':'查询替换', 'map':'Baidu地图', 'gmap':'Google地图',
            'insertvideo':'视频', 'help':'帮助', 'justifyleft':'居左对齐', 'justifyright':'居右对齐', 'justifycenter':'居中对齐',
            'justifyjustify':'两端对齐', 'forecolor':'字体颜色', 'backcolor':'背景色', 'insertorderedlist':'有序列表',
            'insertunorderedlist':'无序列表', 'fullscreen':'全屏', 'directionalityltr':'从左向右输入', 'directionalityrtl':'从右向左输入',
            'RowSpacingTop':'段前距', 'RowSpacingBottom':'段后距','highlightcode':'插入代码', 'pagebreak':'分页', 'insertframe':'插入Iframe', 'imagenone':'默认',
            'imageleft':'左浮动', 'imageright':'右浮动','attachment':'附件', 'imagecenter':'居中', 'wordimage':'图片转存',
            'lineheight':'行间距', 'customstyle':'自定义标题','autotypeset': '自动排版'
        },
        //dialog内容的路径 ～会被替换成URL
        iframeUrlMap:{
            'anchor':'~/dialogs/anchor/anchor.html',
            'insertimage':'~/dialogs/image/image.html',
            'inserttable':'~/dialogs/table/table.html',
            'link':'~/dialogs/link/link.html',
            'spechars':'~/dialogs/spechars/spechars.html',
            'searchreplace':'~/dialogs/searchreplace/searchreplace.html',
            'map':'~/dialogs/map/map.html',
            'gmap':'~/dialogs/gmap/gmap.html',
            'insertvideo':'~/dialogs/video/video.html',
            'help':'~/dialogs/help/help.html',
            'highlightcode':'~/dialogs/code/code.html',
            'emotion':'~/dialogs/emotion/emotion.html',
            'wordimage':'~/dialogs/wordimage/wordimage.html',
            'attachment':'~/dialogs/attachment/attachment.html',
            'insertframe':'~/dialogs/insertframe/insertframe.html',
            'edittd':'~/dialogs/table/edittd.html',
            'snapscreen': '~/dialogs/snapscreen/snapscreen.html'
        },
        //所有的的下拉框显示的内容
        listMap:{
            //字体
            'fontfamily':['宋体', '楷体', '隶书', '黑体', 'andale mono', 'arial', 'arial black', 'comic sans ms', 'impact', 'times new roman'],
            //字号
            'fontsize':[10, 11, 12, 14, 16, 18, 20, 24, 36],
            //段落格式 值:显示的名字
            'paragraph':['p:段落', 'h1:标题 1', 'h2:标题 2', 'h3:标题 3', 'h4:标题 4', 'h5:标题 5', 'h6:标题 6'],
            //段间距 值和显示的名字相同
            'rowspacing':['5', '10', '15', '20', '25'],
            //行内间距 值和显示的名字相同
            'lineheight':['1', '1.5','1.75','2', '3', '4', '5'],
            //block的元素是依据设置段落的逻辑设置的，inline的元素依据BIU的逻辑设置
            //尽量使用一些常用的标签
            //参数说明
            //tag 使用的标签名字
            //label 显示的名字也是用来标识不同类型的标识符，注意这个值每个要不同，
            //style 添加的样式
            //每一个对象就是一个自定义的样式
            'customstyle':[
                {tag:'h1', label:'居中标题', style:'border-bottom:#ccc 2px solid;padding:0 4px 0 0;text-align:center;margin:0 0 20px 0;'},
                {tag:'h1', label:'居左标题', style:'border-bottom:#ccc 2px solid;padding:0 4px 0 0;margin:0 0 10px 0;'},
                {tag:'span', label:'强调', style:'font-style:italic;font-weight:bold;color:#000'},
                {tag:'span', label:'明显强调', style:'font-style:italic;font-weight:bold;color:rgb(51, 153, 204)'}
            ]
        },
        //字体对应的style值
        fontMap:{
            '宋体':['宋体', 'SimSun'],
            '楷体':['楷体', '楷体_GB2312', 'SimKai'],
            '黑体':['黑体', 'SimHei'],
            '隶书':['隶书', 'SimLi'],
            'andale mono':['andale mono'],
            'arial':['arial', 'helvetica', 'sans-serif'],
            'arial black':['arial black', 'avant garde'],
            'comic sans ms':['comic sans ms'],
            'impact':['impact', 'chicago'],
            'times new roman':['times new roman']
        },
        //定义了右键菜单的内容
        contextMenu:[
            {
                label:'删除',
                cmdName:'delete'

            },
            {
                label:'全选',
                cmdName:'selectall'

            },
            {
                label:'删除代码',
                cmdName:'highlightcode',
                icon:'deletehighlightcode'

            },
            {
                label:'清空文档',
                cmdName:'cleardoc',
                exec:function () {

                    if ( confirm( '确定清空文档吗？' ) ) {

                        this.execCommand( 'cleardoc' );
                    }
                }
            },
            '-',
            {
                label:'取消链接',
                cmdName:'unlink'
            },
            '-',
            {
                group:'段落格式',
                icon:'justifyjustify',

                subMenu:[
                    {
                        label:'居左对齐',
                        cmdName:'justify',
                        value:'left'
                    },
                    {
                        label:'居右对齐',
                        cmdName:'justify',
                        value:'right'
                    },
                    {
                        label:'居中对齐',
                        cmdName:'justify',
                        value:'center'
                    },
                    {
                        label:'两端对齐',
                        cmdName:'justify',
                        value:'justify'
                    }
                ]
            },
            '-',
            {
                label:'表格属性',
                cmdName:'edittable',
                exec:function () {
                    this.tableDialog.open();
                }
            },
            {
                label:'单元格属性',
                cmdName:'edittd',
                exec:function () {

                    this.ui._dialogs['tdDialog'].open();
                }
            },
            {
                group:'表格',
                icon:'table',

                subMenu:[
                    {
                        label:'删除表格',
                        cmdName:'deletetable'
                    },
                    {
                        label:'表格前插行',
                        cmdName:'insertparagraphbeforetable'
                    },
                    '-',
                    {
                        label:'删除行',
                        cmdName:'deleterow'
                    },
                    {
                        label:'删除列',
                        cmdName:'deletecol'
                    },
                    '-',
                    {
                        label:'前插入行',
                        cmdName:'insertrow'
                    },
                    {
                        label:'前插入列',
                        cmdName:'insertcol'
                    },
                    '-',
                    {
                        label:'右合并单元格',
                        cmdName:'mergeright'
                    },
                    {
                        label:'下合并单元格',
                        cmdName:'mergedown'
                    },
                    '-',
                    {
                        label:'拆分成行',
                        cmdName:'splittorows'
                    },
                    {
                        label:'拆分成列',
                        cmdName:'splittocols'
                    },
                    {
                        label:'合并多个单元格',
                        cmdName:'mergecells'
                    },
                    {
                        label:'完全拆分单元格',
                        cmdName:'splittocells'
                    }
                ]
            },
            {
                label:'复制(ctrl+c)',
                cmdName:'copy',
                exec:function () {
                    alert( "请使用ctrl+c进行复制" );
                }
            },
            {
                label:'粘贴(ctrl+v)',
                cmdName:'paste',
                exec:function () {
                    alert( "请使用ctrl+v进行粘贴" );
                }
            }
        ],

        initialStyle://编辑器内部样式
        //选中的td上的样式
        '.selectTdClass{background-color:#3399FF !important}' +
        //插入的表格的默认样式
        'table{clear:both;margin-bottom:10px;border-collapse:collapse;word-break:break-all;}' +
        //分页符的样式
        '.pagebreak{display:block;clear:both !important;cursor:default !important;width: 100% !important;margin:0;}' +
        //锚点的样式,注意这里背景图的路径
        '.anchorclass{background: url("' + URL + 'themes/default/images/anchor.gif") no-repeat scroll left center transparent;border: 1px dotted #0000FF;cursor: auto;display: inline-block;height: 16px;width: 15px;}' +
        //设置四周的留边
        '.view{padding:0;word-wrap:break-word;word-break:break-all;cursor:text;height:100%;}\n' +
        'body{margin:8px;font-family:"宋体";font-size:16px;}' +
        //针对li的处理
        'li{clear:both}' +
        //设置段落间距
        'p{margin:5px 0;}',
        //初始化编辑器的内容,也可以通过textarea/script给值，看官网例子
        initialContent:'',
        autoClearinitialContent:false, //是否自动清除编辑器初始内容，注意：如果focus属性设置为true,这个也为真，那么编辑器一上来就会触发导致初始化的内容看不到了
        iframeCssUrl:'themes/default/iframe.css', //要引入css的url
        removeFormatTags:'b,big,code,del,dfn,em,font,i,ins,kbd,q,samp,small,span,strike,strong,sub,sup,tt,u,var', //清除格式删除的标签
        removeFormatAttributes:'class,style,lang,width,height,align,hspace,valign', //清除格式删除的属性
        enterTag:'p', //编辑器回车标签。p或br
        maxUndoCount:20, //最多可以回退的次数
        maxInputCount:20, //当输入的字符数超过该值时，保存一次现场
        selectedTdClass:'selectTdClass', //设定选中td的样式名称
        pasteplain:false, //是否纯文本粘贴。false为不使用纯文本粘贴，true为使用纯文本粘贴
        //提交表单时，服务器获取编辑器提交内容的所用的参数，多实例时可以给容器name属性，会将name给定的值最为每个实例的键值，不用每次实例化的时候都设置这个值
        textarea:'editorValue',
        focus:false, //初始化时，是否让编辑器获得焦点true或false
        indentValue:'2em', //初始化时，首行缩进距离
        pageBreakTag:'_baidu_page_break_tag_', //分页符
        minFrameHeight:320, //最小高度
        autoHeightEnabled:true, //是否自动长高
        autoFloatEnabled:true, //是否保持toolbar的位置不动
        elementPathEnabled:true, //是否启用elementPath
        wordCount:true, //是否开启字数统计
        maximumWords:10000, //允许的最大字符数
        tabSize:4, //tab的宽度
        tabNode:'&nbsp;', //tab时的单一字符
        imagePopup:true, //图片操作的浮层开关，默认打开
        emotionLocalization:false, //是否开启表情本地化，默认关闭。若要开启请确保emotion文件夹下包含官网提供的images表情文件夹
        sourceEditor:"codemirror", //源码的查看方式，codemirror 是代码高亮，textarea是文本框
        tdHeight:'20', //单元格的默认高度
        highlightJsUrl:URL + "third-party/SyntaxHighlighter/shCore.js",
        highlightCssUrl:URL + "third-party/SyntaxHighlighter/shCoreDefault.css",
        codeMirrorJsUrl:URL + "third-party/codemirror2.15/codemirror.js",
        codeMirrorCssUrl:URL + "third-party/codemirror2.15/codemirror.css",
        zIndex : 999, //编辑器z-index的基数
        fullscreen : false, //是否上来就是全屏
        snapscreenHost: '127.0.0.1', //屏幕截图的server端文件所在的网站地址或者ip，请不要加http://
        snapscreenServerFile: URL +"server/upload/php/snapImgUp.php", //屏幕截图的server端保存程序，UEditor的范例代码为“URL +"server/upload/php/snapImgUp.php"”
        snapscreenServerPort: 80,//屏幕截图的server端端口
        snapscreenImgAlign: 'center', //截图的图片默认的排版方式
        snapscreenImgIsUseImagePath: 1, //是否使用上面定义的imagepath，如果为否，那么server端需要直接返回图片的完整路径
        messages:{
            pasteMsg:'编辑器已过滤掉您粘贴内容中不支持的格式！', //粘贴提示
            wordCountMsg:'当前已输入 {#count} 个字符，您还可以输入{#leave} 个字符 ', //字数统计提示，{#count}代表当前字数，{#leave}代表还可以输入多少字符数。
            wordOverFlowMsg:'你输入的字符个数已经超出最大允许值，服务器可能会拒绝保存！', //超出字数限制
            pasteWordImgMsg:'您粘贴的内容中包含本地图片，需要转存后才能正确显示！',
            snapScreenNotIETip: '截图功能需要在ie浏览器下使用',
            snapScreenMsg:'截图上传失败，请检查你的PHP环境。 '
        },
        serialize:function () {                              //配置过滤标签
//            var X = baidu.editor.utils.extend;
//            var inline = {strong:1,em:1,b:1,i:1,u:1,span:1,a:1,img:1};
//            var block = X(inline, {p:1,div:1,blockquote:1,$:{style:1,dir:1}});
            return {
                //编辑器中不能够插入的标签，如果想插入那些标签可以去掉相应的标签名
                blackList:{style:1, link:1, object:1, applet:1, input:1, meta:1, base:1, button:1, select:1, textarea:1, '#comment':1, 'map':1, 'area':1}
//                whiteList: {
//                    div: X(block,{$:{style:1,'class':1}}),
//                    img: {$:{style:1,src:1,title:1,'data-imgref':1, 'data-refid':1, 'class':1}},
//                    a: X(inline, {$:{href:1}, a:0, sup:0}),
//                    strong: inline, em: inline, b: inline, i: inline,
//                    p: block,
//                    span: X(inline, {br:1,$:{style:1,id:1,highlight:1}}
//                }
            };
        }(),
        //下来框默认显示的内容
        ComboxInitial:{
            FONT_FAMILY:'字体',
            FONT_SIZE:'字号',
            PARAGRAPH:'段落格式',
            CUSTOMSTYLE:'自定义样式'
        },
        //自动排版参数
        autotypeset:{
            mergeEmptyline : true,          //合并空行
            removeClass : true,            //去掉冗余的class
            removeEmptyline : false,        //去掉空行
            textAlign : "left",             //段落的排版方式，可以是 left,right,center,justify 去掉这个属性表示不执行排版
            imageBlockLine : 'center',      //图片的浮动方式，独占一行剧中,左右浮动，默认: center,left,right,none 去掉这个属性表示不执行排版
            pasteFilter : false,             //根据规则过滤没事粘贴进来的内容
            clearFontSize : false,           //去掉所有的内嵌字号，使用编辑器默认的字号
            clearFontFamily : false,         //去掉所有的内嵌字体，使用编辑器默认的字体
            removeEmptyNode : false,         // 去掉空节点
            //可以去掉的标签
            removeTagNames : {div:1,a:1,abbr:1,acronym:1,address:1,b:1,bdo:1,big:1,cite:1,code:1,del:1,dfn:1,em:1,font:1,i:1,ins:1,label:1,kbd:1,q:1,s:1,samp:1,small:1,span:1,strike:1,strong:1,sub:1,sup:1,tt:1,u:1,'var':1},
            indent : false,                  // 行首缩进
            indentValue : '2em'             //行首缩进的大小
        }
    };
})();
